cpp/src/parquet/encryption/internal_file_encryptor.cc

```diff
@@ -150,8 +150,8 @@ encryption::AesEncryptor* InternalFileEncryptor::GetMetaAesEncryptor(
   int key_len = static_cast<int>(key_size);
   int index = MapKeyLenToEncryptorArrayIndex(key_len);
   if (meta_encryptor_[index] == nullptr) {
-    meta_encryptor_[index].reset(
-        encryption::AesEncryptor::Make(algorithm, key_len, true, &all_encryptors_));
+    meta_encryptor_[index] = encryption::AesEncryptor::Make(algorithm, key_len, true);
+    all_encryptors_.push_back(meta_encryptor_[index].get());
```

Comment:
Unless I'm mistaken, `all_encryptors_` is never popped from when an encryptor is deallocated. Could we fix this?

Reply:
seems that is owned by `meta_encryptor_` and `data_encryptor_`?

Reply:
All also we can remove `all_encryptors_`, and do:

```
void InternalFileEncryptor::WipeOutEncryptionKeys() {
  properties_->WipeOutEncryptionKeys();

  for (auto const& i : meta_encryptor_) {
    if (i != nullptr) i->WipeOut();
  }
  for (auto const& i : data_encryptor_) {
    if (i != nullptr) i->WipeOut();
  }
}
```

Reply:
Sounds ok to me, though why the nullptr check?

Reply:
`GetMetaAesEncryptor` does a nullable check, I wonder would we face the case that we didn't use it but it being WipeOut

Reply:
```
  // Key must be 16, 24 or 32 bytes in length. Thus there could be up to three
  // types of meta_encryptors and data_encryptors.
  std::unique_ptr<encryption::AesEncryptor> meta_encryptor_[3];
  std::unique_ptr<encryption::AesEncryptor> data_encryptor_[3];
```

And from the comment it might be nullptr

Reply:
I see, SGTM